src/sentry/models/release.py

```diff
@@ -214,6 +219,73 @@ def add_project(self, project):
         else:
             return True
 
+    def set_head_commits(self, head_commits, user, fetch_commits=False):
+        from sentry.models import Commit, ReleaseHeadCommit, Repository
+        from sentry.plugins import bindings
+
+        prev_release = type(self).objects.filter(
+            organization_id=self.organization_id,
+            projects__in=self.projects.all(),
+        ).order_by('-date_added').first()
+
+        commit_list = []
+
+        for head_commit in head_commits:
+            repo = Repository.objects.get_or_create(
+                organization_id=self.organization_id,
+                name=head_commit['repository'],
+            )[0]
+            commit = Commit.objects.get_or_create(
+                organization_id=self.organization_id,
+                repository_id=repo.id,
+                key=head_commit['current_id'],
+            )[0]
+            # update head commit for repo/release if exists
+            ReleaseHeadCommit.objects.create_or_update(
+                organization_id=self.organization_id,
+                repository_id=repo.id,
+                release=self,
+                values={
+                    'commit': commit,
+                }
+            )
+            if fetch_commits:
+                try:
+                    provider_cls = bindings.get('repository.provider').get(repo.provider)
+                except KeyError:
+                    continue
+
+                # if previous commit isn't provided, try to get from
+                # previous release otherwise, give up
+                if head_commit.get('previous_id'):
+                    start_sha = head_commit['previous_id']
+                elif prev_release:
+                    try:
+                        prev_head = ReleaseHeadCommit.objects.get(
```

Comment:
Also, we could refactor this to literally only get the `Commit` instead of doing this to be a bit more optimized.

Something like:

```python
start_sha = Commit.objects.filter(
    releaseheadcommit__organization_id=self.organization_id,
    releaseheadcommit__release=prev_release,
    releaseheadcommit__repository_id=repo.id,
).values_list('key', flat=True)[0]
```

Instead, just to only give you the value you care about.